### PR TITLE
CVSL-1942 adding new endpoint to bulk reset HDC licences

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ services:
       - POSTGRES_USER=licences
       - POSTGRES_DB=hmpps-hdc-api
 
-  localstack-api:
+  localstack-hdc-api:
```
Comment (andrewrlee) [inline docker-compose.yml:16]: fix name conflict with CVL

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/AuditEvent.kt [hunk 2]
```diff
@@ -19,7 +19,8 @@ class AuditEvent(
   @GeneratedValue(strategy = GenerationType.IDENTITY)
   @Column(name = "id", nullable = false)
   val id: Long? = null,
-  var timestamp: LocalDateTime,
+  var timestamp: LocalDateTime?,
+  @Column(name = "`user`", nullable = false)
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/AuditEvent.kt:23]: avoid issue with reserved keyword as table name

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/prison/PrisonApiClient.kt [hunk 3]
```diff
@@ -29,6 +29,29 @@ class PrisonApiClient(@Qualifier("oauthPrisonClient") val prisonerSearchApiWebCl
...
+    return if (response.statusCode().is2xxSuccessful) {
+      log.info("Successfully reset licence for booking $bookingId, status:${response.statusCode()}")
+      true
+    } else {
+      log.warn(
+        "Failed to reset licence for booking $bookingId, status:${response.statusCode()}, uri: ${response.request().uri}, body: ${
+          response.toEntity(String::class.java).block()
+        }",
+      )
+      false
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/prison/PrisonApiClient.kt:51]: an unsuccessful response is logged but doesn't kill the job.. 409 here would mean the record is currently locked

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/softdelete/SoftDeleteService.kt [hunk 4]
```diff
@@ -57,12 +57,6 @@ class SoftDeleteService(
     )
   }
 
-  private fun licencesToMigrate(numberToMigrate: Int): Page<Pair<Licence, Prisoner?>> {
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/softdelete/SoftDeleteService.kt:60]: not used

File: .circleci/config.yml [hunk 5]
```diff
@@ -55,6 +84,10 @@ workflows:
           filters:
             tags:
               ignore: /.*/
+      - integration-test:
+          filters:
+            tags:
+              ignore: /.*/
```
Comment (andrewrlee) [inline .circleci/config.yml:90]: hook in missing tests

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/integration/ResourceSecurityTest.kt [hunk 6]
```diff
@@ -27,6 +27,7 @@ class ResourceSecurityTest : SqsIntegrationTestBase() {
     "GET /v3/api-docs/swagger-config",
     "/error",
     "PUT /queue-admin/retry-all-dlqs",
+    "POST /jobs/populate-deleted-at-for-licences/{numberToProcess}",
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/integration/ResourceSecurityTest.kt:30]: this was failing previously but not part of the build

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/integration/SoftDeleteJobTest.kt [hunk 7]
```diff
@@ -72,15 +71,6 @@ class SoftDeleteJobTest : SqsIntegrationTestBase() {
...
   }
 
-  @Test
-  fun `Unauthorized (401) when no token is supplied`() {
-    webTestClient.get()
-      .uri("/jobs/populate-deleted-at-for-licences/3")
-      .accept(MediaType.APPLICATION_JSON)
-      .exchange()
-      .expectStatus().isEqualTo(HttpStatus.UNAUTHORIZED.value())
-  }
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/integration/SoftDeleteJobTest.kt:82]: this was failing previously - no need for this test as no role required and protected by ingress

**Review by louisenorris — APPROVED** [review]
LGTM

**Pull request merged**